xbdd-web/src/modules/summary/SummaryContainer.jsx

```diff
@@ -5,87 +5,92 @@ import { withStyles } from "@material-ui/core/styles";
 import ProductListContainer from "./productList/ProductListContainer";
 import SummaryStyles from "./styles/SummaryStyles";
 import ProductSummary from "../../models/ProductSummary";
-import { getSummaryOfReports, setProductFavouriteOn, setProductFavouriteOff } from "../../lib/rest/Rest";
+import { getSummaryOfReports, setProductFavouriteOn, setProductFavouriteOff, pinABuild, unPinABuild } from "../../lib/rest/Rest";
 
 class SummaryContainer extends Component {
   constructor(props) {
     super(props);
-    this.state = { productList: null, favouriteList: null };
+    this.state = { productList: null };
 
     this.handleFavouriteChange = this.handleFavouriteChange.bind(this);
+    this.handlePinChange = this.handlePinChange.bind(this);
   }
 
   componentDidMount() {
     getSummaryOfReports().then(summaryData => {
       const productList = new ProductSummary(summaryData).productList;
       this.setState({
-        productList: this.processList(productList, false),
-        favouriteList: this.processList(productList, true),
+        productList: productList,
       });
     });
   }
 
-  processList(productList, isFavouriteList) {
-    const list = [];
-    productList.forEach(element => {
-      const hasBeenAdded = list.some(product => product.name === element.name);
-      const shouldBeAdded = isFavouriteList ? !hasBeenAdded && element.favourite : !hasBeenAdded;
-      if (shouldBeAdded) {
-        list.push(element.clone());
-      }
-    });
-    return list;
-  }
-
   handleFavouriteChange(product) {
     const isFavourite = product.favourite;
-    var newProductList = this.state.productList;
+    const newProductList = this.state.productList;
 
     (isFavourite ? setProductFavouriteOff(product.name) : setProductFavouriteOn(product.name)).then(response => {
       if (response.status === 200) {
-        newProductList.forEach(item => {
-          if (item.name === product.name) {
-            item.favourite = !isFavourite;
-          }
+        const newProduct = newProductList.find(item => item.name === product.name);
+        newProduct.setFavouriteStatus(!isFavourite);
+        this.setState({
+          productList: newProductList,
         });
+      }
+    });
+  }
+
+  handlePinChange(event, product, version, build, isPinned) {
+    const newProductList = this.state.productList;
+
+    (isPinned
```

Comment:
this ternary makes the workings of this not particularly obvious. Can we pull that out into a function that returns a promise?